server/assertions/assertions.go

```diff
@@ -57,10 +57,11 @@ func assertIndividualSpans(a model.Assertion, spans []traces.Span) model.Asserti
 	res := make([]model.SpanAssertionResult, len(spans))
 	allPassed := true
 	for i, span := range spans {
+		spanID := span.ID
 		res[i] = apply(
 			a,
 			span.Attributes.Get(a.Attribute.String()),
-			&span.ID,
```

Comment:
As `span` is a single variable that changes its value every time we iterate through spans and we are passing its address, the persisted value was the ID of the last span in `spans`.

By storing the id in a local variable inside the loop, we fix this problem. This new variable is reallocated every time the loop executes.